src/components/Shop.jsx

```diff
@@ -0,0 +1,34 @@
+import styled from "styled-components";
+import Intro from "./Intro";
+import ShoppingList from "./ShoppingList";
+import ToggleBtn from "./ToggleBtn";
+import { useEffect, useState } from "react";
+
+export default function Shop() {
+  const [items, setItems] = useState([]);
+  const [isSale, setIsSale] = useState(false);
+  const url = !isSale ? "api/products.json" : "api/sale_products.json";
+  const buttonText = !isSale ? "Show Sale product" : "All products";
+
+  useEffect(() => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => setItems(data));
+  }, [isSale]);
+
+  const buttonHandler = () => setIsSale(!isSale);
```

Comment:
```suggestion
  const buttonHandler = () => setIsSale(prev => !prev);
```
react에서 같은 값을 state를 다룰 때는 항상 [previous state](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state)로 다루는 것을 추천드려요!!
추가로... 저는 의미론적인 부여를 위해 toggleSaleHandler라는 이름이 더 좋지 않을까? 라는 생각을 했었어요.